river_dl/preproc_utils.py

```diff
@@ -123,18 +123,23 @@ def split_into_batches(data_array, seq_len=365, offset=1.0):
     :param data_array: [numpy array] array of training data with dims [nseg,
     ndates, nfeat]
     :param seq_len: [int] length of sequences (e.g., 365)
-    :param offset: [float] 0-1, how to offset the batches (e.g., 0.5 means that
-    the first batch will be 0-365 and the second will be 182-547)
+    :param offset: [float] How to offset the batches. Values < 1 are taken as fractions, (e.g., 0.5 means that
+    the first batch will be 0-365 and the second will be 182-547), values > 1 are used as a constant number of
+    observations to offset by.
     :return: [numpy array] batched data with dims [nbatches, nseg, seq_len
     (batch_size), nfeat]
     """
-    combined = []
-    for i in range(int(1 / offset)):
-        start = int(i * offset * seq_len)
-        idx = np.arange(start=start, stop=data_array.shape[1] + 1, step=seq_len)
-        split = np.split(data_array, indices_or_sections=idx, axis=1)
-        # add all but the first and last batch since they will be smaller
-        combined.extend([s for s in split if s.shape[1] == seq_len])
+    if offset>1:
+        period = offset
+    else:
+        period = int(offset*seq_len)
+    num_batches = int(data_array.shape[1]//period)
```

Comment:
Cool using `//`! I think I learned about that a long time ago but I'd forgotten about it. 

I think with that, it returns an `int` so we won't need to use the `int()` as well.

```suggestion
    num_batches = data_array.shape[1]//period
```

Reply:
Derp! Thanks for catching that.